src/transformStateWithClones.js

```diff
@@ -7,7 +7,50 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const transformState = [];
+  const cloneState = { ...state };
+
+  for (const action of actions) {
+    let newObj = {};
+
+    switch (action.type) {
+      case 'clear':
+        newObj = { ...clear(cloneState) };
+        break;
+
+      case 'addProperties':
+        newObj = { ...addProperties(cloneState, action.extraData) };
+        break;
+
+      case 'removeProperties':
+        newObj = { ...removeProperties(cloneState, action.keysToRemove) };
+        break;
+    }
```

Comment:
The 'switch' statement is missing a 'default' case. It is good practice to include a 'default' case to handle unexpected 'action.type' values.